diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +121,181 @@ class LineLength : Rule("line-length") {
+                }
+            }
+            node.hasChildOfType(PARENTHESIZED) -> {
+                node.getAllChildrenWithType(PARENTHESIZED).forEach {
+                    searchBinaryExpression(it, binList)
+                }
+            }
+            else -> {
+                binList.add(node)
+                binList.add(node.treeParent.findChildByType(PREFIX_EXPRESSION) ?: return)
+            }
+        }
+    }
+
+    private fun dfsForProperty(node: ASTNode, binList: MutableList<ASTNode>) {
+        node.getChildren(null).forEach {
+            if (it.elementType in PROPERTY_LIST)
+                binList.add(it)
+            dfsForProperty(it, binList)
+        }
+    }
+
+    private fun fixProperty(parent: ASTNode, configuration: LineLengthConfiguration){
+        if (!parent.hasChildOfType(STRING_TEMPLATE)) {
+            if (parent.hasChildOfType(BINARY_EXPRESSION)) {
+                val leftOffset = positionByOffset(parent.findChildByType(BINARY_EXPRESSION)!!.startOffset).second
+                fixCondition(parent, configuration, leftOffset, true)
+            }
+            if (parent.hasChildOfType(PARENTHESIZED)){
+                val newParent = parent.findChildByType(PARENTHESIZED)!!
+                val leftOffset = positionByOffset(newParent.findChildByType(BINARY_EXPRESSION)!!.psi.textOffset).second
+                fixCondition(newParent, configuration, leftOffset)
+            }
+        }
+        else
+            createNodes(parent, configuration)
+    }
+
+    private fun createNodes(parent: ASTNode, configuration: LineLengthConfiguration) {
+        var text = parent.findChildByType(STRING_TEMPLATE)!!.text
+        // trim to remove first and last quote
+        text = text.trimStart(text.first()).trimEnd(text.last())
+        val col = positionByOffset(parent.findChildByType(STRING_TEMPLATE)!!.psi.textOffset).second
+        parent.removeChild(parent.findChildByType(STRING_TEMPLATE)!!)
+        val correctTextList = mutableListOf<String>().apply {
+            add(text.substring(0, configuration.lineLength.toInt() - col - FIRST_STRING_PART_OFFSEST))
+            text = text.substring(configuration.lineLength.toInt() - col - FIRST_STRING_PART_OFFSEST)
+            addAll(text.chunked(configuration.lineLength.toInt() - REST_STRING_PART_OFFEST))
+        }
+                .toList()
+        var prevExp = CompositeElement(BINARY_EXPRESSION)
+        parent.addChild(prevExp, null)
+        correctTextList.reversed().dropLast(1).forEach { textBinExpress ->
+            createNodesBetweenStringTemplates(prevExp)
+            createStringTemplate(textBinExpress, prevExp)
+            val newExp = CompositeElement(BINARY_EXPRESSION)
+            prevExp.addChild(newExp, prevExp.firstChildNode)
+            prevExp = newExp
+        }
+        createStringTemplate(correctTextList.first(), prevExp)
+    }
+
+    private fun createStringTemplate(text: String, prevExp: CompositeElement) {
```

Comment:
Do you have any tests for string templates - with variables inside, line `"string $foo"`? What will happen if the end of maximum line length falls inside variable used in the string?